### ISSUE TITLE
OnCanvasTouch Function?

### ISSUE THREAD

**Issue by devinyim** [namespace-ee/react-calendar-timeline#168]
Hello,

I'd like to know if it is possible to trigger an event when you **touch** an empty canvas cell. I've tried using onCanvasClick which only works when you **click** on an empty canvas cell.

**Comment by kmueller** [comment]
Give the following workaround a try:

In Timeline.js replace function touchEnd (line 440...)

```
  touchEnd = (e) => {
    if (this.lastTouchDistance) {
      e.preventDefault()

      this.lastTouchDistance = null
    }
    if (this.lastSingleTouch) {
      e.preventDefault()

      this.lastSingleTouch = null
      this.singleTouchStart = null
    }
  }

```
by this code:

```
  touchEnd = (e) => {
    if (this.lastTouchDistance) {
      e.preventDefault()

      this.lastTouchDistance = null
    }
    if (this.lastSingleTouch) {
      e.preventDefault()

            // Ensure that no touchmove was effected
            if( (this.lastSingleTouch.x == this.singleTouchStart.x)
             && (this.lastSingleTouch.y == this.singleTouchStart.y) ) {
                
                // Add clientX and clientY position to event
                // so that groupId and time can be calculated
                e.clientX = this.lastSingleTouch.x; 
                e.clientY = this.lastSingleTouch.y;  
                
                // Find out if tap was in scrollarea (below header) or not
                var wasScrollAreaClick = false;
                var parentPosition = (0, _utils.getParentPosition)(e.currentTarget);
                var screenY = this.lastSingleTouch.screenY;
                var headerHeight = this.props.headerLabelGroupHeight + this.props.headerLabelHeight;
                var stickyOffset = this.props.stickyOffset;                
                var realOffset = parentPosition.y - screenY; 
                if (realOffset > 0) {
                    wasScrollAreaClick = e.clientY > realOffset + headerHeight;
                } else {
                    wasScrollAreaClick = e.clientY > stickyOffset + headerHeight;
                }
                
                // Call appropriate actions
                if (wasScrollAreaClick) {                  
                    this.scrollAreaClick(e);
                }
            } 

      this.lastSingleTouch = null
      this.singleTouchStart = null
    }
  }
```

**Comment by mcMickJuice** [comment]
Fixed in v0.16.3
